DigitalLearningSolutions.Data/Services/TrackerService.cs

```diff
@@ -0,0 +1,43 @@
+﻿namespace DigitalLearningSolutions.Data.Services
+{
+    using System;
+    using DigitalLearningSolutions.Data.Enums;
+    using DigitalLearningSolutions.Data.Models.Tracker;
+    using Microsoft.Extensions.Logging;
+
+    public interface ITrackerService
+    {
+        string ProcessQuery(TrackerEndpointQueryParams query);
+    }
+
+    public class TrackerService : ITrackerService
+    {
+        private readonly ILogger<TrackerService> logger;
+
+        public TrackerService(ILogger<TrackerService> logger)
+        {
+            this.logger = logger;
+        }
+
+        public string ProcessQuery(TrackerEndpointQueryParams query)
+        {
+            if (string.IsNullOrWhiteSpace(query.Action))
+            {
+                return TrackerEndpointErrorResponse.NullAction;
+            }
+            
+            try
+            {
+                return query.Action! switch

```

Comment:
When we change TrackerEndpointAction into an enum, we'll have to parse it manually sadly. Could you do that before entering the switch statement? Case should be ignored.